docs/development/best-practices.md

```diff
@@ -3,6 +3,26 @@
 This document explains our best practices.
```

Comment:
I thought of putting the branch name stuff in the `local-development.md` and `remote-development.md` docs. But then I would have the same information in two files.

So I'm putting the branch name stuff in `best-practices.md` instead. 😄